src/views/activity/activity_item_view.js

```diff
@@ -20,5 +26,26 @@ export default BaseView.extend({
 
   sendSelectionDetails (event) {
     webChannel.sendUrlSelected(this.model.url, 'url');
+  },
+
+  injectFavicon () {
+    // Manually construct the favicon element and inject it favicon into the
+    // DOM, allowing us to delegate events to its onLoad event, which is not
+    // possible through Ampersand since load events do not bubble.
+    const favicon = document.createElement('img');
+    favicon.src = this.model.displayImage();
+    favicon.onload = this.calculateIconColor;
```

Comment:
Oh also, if we're creating the Image in JS, we'll need to shove it into the DOM as part of the onload handler.